src/contracts/libraries/GPv2Encoding.sol

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: LGPL-3.0-or-newer
+pragma solidity ^0.6.12;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+/// @title Gnosis Protocol v2 Encoding Library.
+/// @author Gnosis Developers
+library GPv2Encoding {
+    /// @dev A struct representing an executed order that can be used for
+    /// settling a batch.
+    struct Order {
+        address owner;
+        IERC20 sellToken;
+        IERC20 buyToken;
+        uint256 sellAmount;
+        uint256 buyAmount;
+        uint32 validTo;
+        uint32 nonce;
+        uint256 tip;
+        OrderKind kind;
+        bool partiallyFillable;
+        uint256 executedAmount;
+        uint8 sellTokenIndex;
+        uint8 buyTokenIndex;
+        bytes32 digest;
+    }
+
+    /// @dev An enum describing an order kind, either a buy or a sell order.
+    enum OrderKind {Sell, Buy}
+
+    /// @dev The stride of an encoded order.
+    uint256 internal constant ORDER_STRIDE = 204;
+
+    /// @dev Bit position for the [`OrderKind`] encoded in the flags.
+    uint8 internal constant ORDER_KIND_BIT = 0;
+    /// @dev Bit position for the `partiallyFillable` value encoded in the
+    /// flags.
+    uint8 internal constant ORDER_PARTIALLY_FILLABLE_BIT = 1;
+
+    /// @dev Decodes a signed order from calldata bytes into memory.
+    /// Orders are tightly packed in order to reduce calldata and associated gas
+    /// costs. They contain the following fields:
+    /// ```
+    /// struct Order {
```

Comment:
Can you rename this to something else than `Order`? I was checking the `mstore` operation and it appeared like it overwrites `sellTokenIndex` and `sellTokenIndex`. Instead, it was using a different `Order` struct.

Reply:
`EncodedOrder`?